lib/internal/errors.js

```diff
@@ -21,9 +21,17 @@ const { kMaxLength } = internalBinding('buffer');
 
 const MainContextError = Error;
 const ErrorToString = Error.prototype.toString;
-// Polyfill of V8's Error.prepareStackTrace API.
-// https://crbug.com/v8/7848
+const internalOverriddenErrors = new WeakMap();
 const prepareStackTrace = (globalThis, error, trace) => {
+  // API for node internals to override error stack formatting.
+  if (internalOverriddenErrors.has(error)) {
+    const f = internalOverriddenErrors.get(error);
```

Comment:
Nit: maybe use at least `fn` to indicate it's a function (that's how we mostly named them throughout the code)?